great_expectations/rule_based_profiler/data_assistant/data_assistant_dispatcher.py

```diff
@@ -0,0 +1,56 @@
+from typing import Dict, Optional, Type
+
+from great_expectations.expectations.registry import get_data_assistant_impl
+from great_expectations.rule_based_profiler.data_assistant import DataAssistant
+from great_expectations.rule_based_profiler.data_assistant.data_assistant_runner import (
+    DataAssistantRunner,
+)
+
+
+class DataAssistantDispatcher:
+    """
+    DataAssistantDispatcher intercepts requests for "DataAssistant" classes by their registered names and manages their
+    associated "DataAssistantRunner" objects, which process invocations of calls to "DataAssistant" "run()" methods.
+    """
+
+    DATA_ASSISTANT_TYPE_ALIASES: Dict[str, str] = {
+        "volume": "volume_data_assistant",
+    }
+
+    def __init__(self, data_context: "BaseDataContext"):  # noqa: F821
+        """
+        Args:
+            data_context: BaseDataContext associated with DataAssistantDispatcher
+        """
+        self._data_context = data_context
+
+        self._data_assistant_runner_cache = {}
+
+    def __getattr__(self, name):
+        # Both, full registered "volume_data_assistant" data_assistant_type and alias name are supported for invocation.
+        data_assistant_cls: Type["DataAssistant"] = get_data_assistant_impl(
+            data_assistant_type=name.lower()
+        ) or get_data_assistant_impl(
+            data_assistant_type=DataAssistantDispatcher.DATA_ASSISTANT_TYPE_ALIASES.get(
+                name.lower(), ""
+            )
+        )
```

Comment:
Nitpick - Could we split up this logic out of a single statement? There's a lot going on here.

Additionally, I don't think `Type["DataAssistant"]` is accurate if we possibly return `""`

Reply:
@cdkini Fixed -- thanks!